### PR TITLE
Use a synthesized local for the lockTaken parameter and create a region for it

### PR DESCRIPTION
**Pull request by 333fred**
@dotnet/roslyn-compiler @AlekseyTs @dotnet/analyzer-ioperation for review. This is a small change to remove a confusing scenario in lock/SyncLock statements where a flow capture reference was used without a corresponding flow capture.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
>                 enterMethod = (IMethodSymbol)_compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_Threading_Monitor__Enter);

Consider asserting that baseLockStatement.LockTakenSymbol is null

---
Refers to: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs:3261 in 963ad99. [](commit_id = 963ad99c40d007f43a8fa62fe9116a901e427c0c, deletion_comment = False)

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: >isDeclaration: false [](start = 92, length = 20)

Should this be true?

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 1)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need second review.

**Review by jaredpar — APPROVED** [review]
:shipit:

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 2)

**Review by mavasani — APPROVED** [review]

**Pull request merged**
